format.go

```diff
@@ -0,0 +1,151 @@
+package cid
+
+import (
+	"bytes"
+	"fmt"
+
+	mb "github.com/multiformats/go-multibase"
+	mh "github.com/multiformats/go-multihash"
+)
+
+// FormatRef is a string documenting the format string for the Format function
+const FormatRef = `
+   %% literal %
+   %b multibase name
+   %B multibase code
+   %v version string
+   %V version number
+   %c codec name
+   %C codec code
+   %h multihash name
+   %H multihash code
+   %L hash digest length
+   %m multihash encoded in base %b (with multibase prefix)
+   %M multihash encoded in base %b without multibase prefix
+   %d hash digest encoded in base %b (with multibase prefix)
+   %D hash digest encoded in base %b without multibase prefix
+   %s cid string encoded in base %b (1)
+   %S cid string encoded in base %b without multibase prefix
+   %P cid prefix: %v-%c-%h-%L
+
+(1) For CID version 0 the multibase must be base58btc and no prefix is
+used.  For Cid version 1 the multibase prefix is included.
+`
+
+// Format formats a cid according to the format specificer as
+// documented in the FormatRef constant
+func Format(fmtStr string, base mb.Encoding, cid *Cid) (string, error) {
+	p := cid.Prefix()
+	out := new(bytes.Buffer)
```

Comment:
No need to allocate here. We can just put this on the stack. It even has a small internal 64byte buffer to avoid allocations when formatting small things.

Reply:
Okay fixed.